tests/helper/helper_http.go

```diff
@@ -51,24 +51,14 @@ func HttpWaitForWithStatus(url string, match string, maxRetry int, interval int,
 	Fail(fmt.Sprintf("Failed after %d retries. Content in %s doesn't include '%s'.", maxRetry, url, match))
 }
 
-// GetRandomFreePort returns a random free port(string) between 1024 and 65535, or within a given portRange if provided
-// WARN: If length of portRange is anything other than 2 and first index is greater-than-or-equal-to second index, it will use the default range.
-func GetRandomFreePort(portRange ...int) string {
-	max := 65535
-	min := 1024
-	var (
-		startPort = rand.Intn(max-min) + min // #nosec  // cannot use crypto/rand library here
-		endPort   = max
-	)
-	// WARN: If length of portRange is anything other than 2 and first index is gte second index, it will be ignored
-	if len(portRange) == 2 && portRange[0] < portRange[1] {
-		startPort = portRange[0]
-		endPort = portRange[1]
-	}
-	freePort, err := util.NextFreePort(startPort, endPort, nil)
-	if err != nil {
-		Fail("failed to obtain a free port")
-	}
-	return strconv.Itoa(freePort)
+var startPort int64 = 30000
 
+// GetRandomFreePort increases the counter of global variable startPort, checks if the counter value is a free port, and returns.
+// If the port is not free, it calls itself again. Checking the port is an additional validation to ensure we don't run into random failures because of port unavailability.
+func GetRandomFreePort() string {
+	atomic.AddInt64(&startPort, 1)
+	if !util.IsPortFree(int(startPort)) {
+		return GetRandomFreePort()
+	}
```

Comment:
I'm not sure IsPortFree is atomic here. You could just expect the port is free.

The random ports are in the range of the Ephemeral ports, in my machine it is:

```
$ sysctl net.ipv4.ip_local_port_range
net.ipv4.ip_local_port_range = 32768	60999
```

So we can consider ports below 32768 will be free.